specification/cognitiveservices/OpenAI.Inference/models/azure.cognitive.search.options.tsp

```diff
@@ -0,0 +1,104 @@
+import "@typespec/versioning";
+import "./azure.chat.extensions.tsp";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+
+@doc("Optional settings to control how fields are processed when using a configured Azure Cognitive Search resource.")
+@added(ServiceApiVersions.v2023_08_01_Preview)
+model AzureCognitiveSearchIndexFieldMappingOptions {
+  @doc("The name of the index field to use as a title.")
+  @projectedName("csharp", "TitleFieldName")
+  titleField?: string;
+
+  @doc("The name of the index field to use as a URL.")
+  @projectedName("csharp", "UrlFieldName")
+  urlField?: string;
+
+  @doc("The name of the index field to use as a filepath.")
+  @projectedName("csharp", "FilepathFieldName")
+  filepathField?: string;
+
+  @doc("The names of index fields that should be treated as content.")
+  @projectedName("csharp", "ContentFieldNames")
+  contentFieldNames?: string[];
+
+  @doc("The separator pattern that content fields should use.")
+  @projectedName("csharp", "ContentFieldSeparator")
+  contentFieldSeparator?: string;
+
+  @doc("The names of fields that represent vector data.")
+  @projectedName("csharp", "VectorFieldNames")
+  vectorFields?: string[];
+}
+
+@doc("""
+The type of Azure Cognitive Search retrieval query that should be executed when using it as an Azure OpenAI chat extension.
+""")
+@added(ServiceApiVersions.v2023_08_01_Preview)
+enum AzureCognitiveSearchQueryType {
+  @doc("Represents the default, simple query parser.")
+  simple: "simple",
+
+  @doc("Represents the semantic query parser for advanced semantic modeling.")
+  semantic: "semantic",
+
+  @doc("Represents vector search over computed data.")
+  vector: "vector",
+
+  @doc("Represents a combination of the simple query strategy with vector data.")
+  vectorSimpleHybrid: "vectorSimpleHybrid",
+
+  @doc("Represents a combination of semantic search and vector data querying.")
+  vectorSemanticHybrid: "vectorSemanticHybrid",
+}
+
+@added(ServiceApiVersions.v2023_08_01_Preview)
+@doc("""
+A specific representation of configurable options for Azure Cognitive Search when using it as an Azure OpenAI chat
+extension.
+""")
+model AzureCognitiveSearchChatExtensionConfiguration {
+  @doc("""
+  The type label to use when configuring Azure OpenAI chat extensions. This should typically not be changed from its
+  default value for Azure Cognitive Search.
+  """)
+  type: AzureChatExtensionType.azureCognitiveSearch;
+
+  @doc("The absolute endpoint path for the Azure Cognitive Search resource to use.")
+  @projectedName("csharp", "SearchEndpoint")
+  endpoint: url;
+
+  @doc("The API key to use with the specified Azure Cognitive Search endpoint.")
+  @projectedName("csharp", "SearchKey")
```

Comment:
wondering if we can document it as "admin key", not "query key" because there are 2 types of keys in azure search